syllabus/core/stat_recorder.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+from syllabus.task_space import TaskSpace
+from gymnasium.spaces import Discrete #, MultiDiscrete?
+import json
+import os
+
+def convert_numpy(obj):
+    if isinstance(obj, np.generic):
+        return obj.item()  # Use .item() to convert numpy types to native Python types
+    raise TypeError
+
+class StatRecorder:
+    """
+    Individual stat tracking for each task.
+    """
+
+    def __init__(self, task_space: TaskSpace):
+        """Initialize the StatRecorder"""
+
+        self.write_path = '/Users/allisonyang/Downloads'
+
+        self.task_space = task_space
+
+        assert isinstance(self.task_space, TaskSpace), f"task_space must be a TaskSpace object. Got {type(task_space)} instead."
+        assert isinstance(self.task_space.gym_space, Discrete), f"assume that task_space should be of type Discrete"
```

Comment:
I appreciate the type checking :)
```suggestion
        assert isinstance(self.task_space.gym_space, Discrete), f"Only Discrete task spaces are supported. Got {type(task_space.gym_space)}"
```